File: src/api/available_restaurants/response.rs
use crate::models::Restaurant;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
struct RestaurantsResponseCollection {
  pub restaurants: Vec<Restaurant>,
}

pub fn parse_response(response: String) -> Vec<Restaurant> {
  let json: String = response.chars().filter(|c| !c.is_control()).collect();
  let collection: RestaurantsResponseCollection = serde_json::from_str(&json).unwrap();

  collection.restaurants
}

File: src/api/availableRestaurants/response.ts
import type { Response } from "@literate.ink/utilities";
import { type Restaurant, type RestaurantAPI, translateRestaurantFromAPI } from "~/models/Restaurant";

interface RestaurantsResponseCollection {
  restaurants: Array<RestaurantAPI>
}

export function parseResponse(response: Response): Array<Restaurant> {
  const content = response.content.replace(/[\u0000-\u001F]/g, "");
  const json = JSON.parse(content) as RestaurantsResponseCollection;

  return json.restaurants.map(translateRestaurantFromAPI);
}
